modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/PageMemoryEx.java

```diff
@@ -177,9 +178,12 @@ public void checkpointWritePage(
     public FullPageId pullPageFromCpBuffer();
 
     /**
-     * Calculates throttling condition.
+     * Whether Checkpoint Buffer is currently in the danger zone.
+     *
+     * @return {@code true} if measures like throttling to protect Checkpoint Buffer should be applied,
+     * and {@code false} otherwise.
      */
-    public boolean shouldThrottle();
+    public boolean isCPBufferInDangerZone();
```

Comment:
This name isn't very informative but is scary. I suggest to change it to more neutral `isCPBufferOverflowThresholdExceeded`.

Reply:
Renamed the method to `isCpBufferOverflowThresholdExceeded` (with 'Cp' instead of 'CP' because there is a method called `pullPageFromCpBuffer` in this interface.